R/bs-theme-preset.R

```diff
@@ -66,11 +61,34 @@ bs_preset_bundle.default <- function(preset, ...) {
 
 #' @export
 bs_preset_bundle.bs_preset <- function(preset, ...) {
-  switch(
-    preset$type %||% "",
-    builtin = builtin_bundle(preset$name, version = preset$version),
-    bootswatch = bootswatch_bundle(preset$name, version = preset$version),
-    NextMethod()
+  NextMethod()
+}
+
+#' @export
+bs_preset_bundle.bs_preset_builtin <- function(preset, ...) {
+  stopifnot(inherits(preset, "bs_preset"))
+  builtin_bundle(preset$name, version = preset$version)
+}
+
+#' @export
+bs_preset_bundle.bs_preset_bootswatch <- function(preset, ...) {
+  stopifnot(inherits(preset, "bs_preset"))
```

Comment:
Shouldn't this be `bs_theme_with_preset`? I think I'd also prefer to not have the `stopifnot()`s and get rid of `bs_preset_bundle.bs_preset()`?

Reply:
Also, now that I think through this a bit more, I'm questioning the utility of using S3 classes at all. Couldn't we get rid of all of them and `bs_preset_bundle()` could just be:

```r
bs_preset_bundle <- function(version, preset, bootswatch) {
  preset <- resolve_bs_preset(preset, bootswatch, version = version)
  switch(
    preset$type, 
    builtin = builtin_bundle(preset$name, version = preset$version),
    bootswatch = bootswatch_bundle(preset$name, version = preset$version)
  )
}
```

This would also have the benefit of making the `bs_theme_update()` cleaner (since wouldn't need to `grep()` for S3 classes)

Reply:
I updated the PR to only add the `bs_theme_with_preset` class to the theme object. I also renamed `preset$class` to `preset$theme_class` and added comments in `new_bs_preset()` for clarity.